### PR TITLE
[D1X] Add interstitials, component tweaks, placeholders

### PR DESCRIPTION
**Pull request by estrattonbailey**
Adds interstitial components, but does not integrate them just yet.

This PR touches the `FeedCard` and `ProfileCard` a little also to get them to behave well in varied use like this.

Note: if these requests fail for whatever reason, we simply don't render anything.
Note note: opted not to allow pinning a feed from the interstitial because at present it reloads the page and you lose your position.

Testing:
Just plop `posts/Feed.tsx` [from my other PR](https://github.com/bluesky-social/social-app/pull/4698/files#diff-baa45c337074da7a535186a1109496b9c50eb77de34557409ed4f72ad0af5d45) into your local setup to get them rendering. You'll also need to add your ID to StatSig overrides.
- check Feeds screen
  - web desktop/mobile
  - native
- check Search screen, same thing
- and check Discover/Following feeds to view these in action

CleanShot 2024-06-28 at 10 45 08@2x
CleanShot 2024-06-28 at 10 21 20@2x
CleanShot 2024-06-28 at 10 21 28@2x
CleanShot 2024-06-28 at 10 22 01@2x
CleanShot 2024-06-28 at 10 22 11@2x
CleanShot 2024-06-28 at 10 22 19@2x
CleanShot 2024-06-28 at 10 22 33@2x
CleanShot 2024-06-28 at 10 22 50@2x
CleanShot 2024-06-28 at 10 22 54@2x
CleanShot 2024-06-28 at 10 37 13@2x

### CODE REVIEW

**Comment by pfrazee** [comment]
WDYT about always grabbing 4 suggestions, then on web doing a 2x2 grid so that the width isn't so awkward?

**Comment by estrattonbailey** [comment]
CleanShot 2024-07-01 at 15 18 53@2x
CleanShot 2024-07-01 at 15 19 00@2x
CleanShot 2024-07-01 at 15 18 49@2x
CleanShot 2024-07-01 at 15 13 49@2x

**Comment by estrattonbailey** [comment]
CleanShot 2024-07-01 at 18 41 26@2x

File: src/components/FeedInterstitials.tsx [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+      style={[a.border_t, t.atoms.border_contrast_low, t.atoms.bg_contrast_25]}>
+      <View style={[a.pt_2xl, a.px_lg, a.flex_row, a.pb_xs]}>
+        <Text
+          style={[
+            a.flex_1,
+            a.text_lg,
+            a.font_bold,
+            t.atoms.text_contrast_medium,
+          ]}>
+          Suggested for you
```
Comment (surfdude29) [inline src/components/FeedInterstitials.tsx:175]: Would it be possible to mark this string for localization please? 🙏

File: src/components/FeedInterstitials.tsx [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+      style={[a.border_t, t.atoms.border_contrast_low, t.atoms.bg_contrast_25]}>
+      <View style={[a.pt_2xl, a.px_lg, a.flex_row, a.pb_xs]}>
+        <Text
+          style={[
+            a.flex_1,
+            a.text_lg,
+            a.font_bold,
+            t.atoms.text_contrast_medium,
+          ]}>
+          Some other feeds you might like
```
Comment (surfdude29) [inline src/components/FeedInterstitials.tsx:303]: And this one too please? :)

File: src/components/FeedInterstitials.tsx (hunk 1 above)
Comment (estrattonbailey) [reply src/components/FeedInterstitials.tsx]: Thanks!

**Review by gaearon — APPROVED** [review]

**Pull request merged**